packages/core-data/src/utils/crdt-blocks.ts

```diff
@@ -187,3 +195,37 @@ export function mergeBlocks(
 		knownClientIds.add( clientId );
 	}
 }
+
+/**
+ * Determine if a block should be synced.
+ *
+ * Ex: A gallery block should not be synced until the images have been
+ * uploaded to WordPress, and their url is available. Before that,
+ * it's not possible to access the blobs on a client as those are
+ * local.
+ *
+ * @param block The block to check.
+ * @return True if the block should be synced, false otherwise.
+ */
+function shouldBlockBeSynced( block: Block ): boolean {
+	// Verify that the gallery block is ready to be synced.
+	// This means that, all images have had their blobs converted to full URLs.
+	// Checking for only the blobs ensures that blocks that have just been inserted work as well.
+	if ( 'core/gallery' === block.name ) {
```

Comment:
I switched this from a switch case into an if statement just to keep it clean.

The logic is as follows:

- All core blocks, save for the gallery block will always be synced.
- The gallery block will only be synced once there are no blobs in it.
- Any third party can be customize its block syncing process.